lib/ansible/modules/crypto/openssl_privatekey.py

```diff
@@ -136,9 +147,26 @@ def __init__(self, module):
         self.path = module.params['path']
         self.mode = module.params['mode']
         self.changed = True
+        self.privatekey = None
+        self.fingerprint = {}
         self.check_mode = module.check_mode
 
 
+    def get_fingerprint(self):
+        """Generate the fingerprint of the public key. """
+
+        fingerprint = {}
+        privatekey = crypto.load_privatekey(crypto.FILETYPE_PEM, open(self.path, 'r').read())
+        publickey_content = crypto.dump_publickey(crypto.FILETYPE_PEM, privatekey)
+        pubkeyb64 = publickey_content.replace('-----BEGIN PUBLIC KEY-----\n', '').replace('-----END PUBLIC KEY-----\n', '')
```

Comment:
Are you certain that using `crypto.dump_publickey`, stripping the PEM fences and Base64-deconding produces (byte for byte) the DER encoding?

Also, pyOpenSSL [can produce DER](http://www.pyopenssl.org/en/stable/api/crypto.html#OpenSSL.crypto.FILETYPE_ASN1) without additional wrangling, according to the documentation.